proto/schedule/utils.go

```diff
@@ -0,0 +1,94 @@
+package schedule
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	"io/ioutil"
+	"time"
+)
+
+const (
+	DateFormat              = "Mon 02 Jan 2006"
+	defaultStopTimeDays 	= 7
+	dayDuration             = 24 * time.Hour
+	defaultStopTimeDuration = defaultStopTimeDays * dayDuration
+)
+
+func (s *Schedule) Validate() error {
+	if s == nil {
+		return fmt.Errorf("nil Schedule")
+	}
+
+	if s.Shifts == nil || len(s.Shifts) == 0 {
+		fmt.Printf("No input shifts found. Starting from scratch.")
+		return nil
+	}
+
+	for i, shift := range s.Shifts {
```

Comment:
Also validate that each `StartDate` is later than the previous `StartDate`

Reply:
Done.